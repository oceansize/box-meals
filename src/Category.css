.category {
  border: 1px solid salmon;
  display: inline;
  margin-right: 1em;
}
