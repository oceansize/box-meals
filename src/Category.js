import React, { Component } from 'react';
import './Category.css';

class Category extends Component {
  render() {
    return(
      <li className="category">{this.props.name}</li>
    );
  }
}

export default Category;

