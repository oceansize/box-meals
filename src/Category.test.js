import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Category from './Category';

describe(Category, () => {

  it('matches the component snapshot', () => {
    const tree = renderer.create(
      <Category name="Desserts" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays the correct name', () => {
    const categoryName = 'Sauces & Dressings';
    const component = shallow(
      <Category name="Sauces & Dressings" />
    );
    expect(component.text()).toContain(categoryName);
  });
});

