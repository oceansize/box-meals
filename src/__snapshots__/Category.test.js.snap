// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class Category extends _react.Component {
  render() {
    return (
      _react2.default.createElement('li', { className: 'category', __source: { fileName: _jsxFileName, lineNumber: 7 }, __self: this }, this.props.name));

  }} matches the component snapshot 1`] = `
<li
  className="category"
>
  Desserts
</li>
`;
